main.cpp

```diff
@@ -324,20 +342,49 @@ int main(int argc, const char *argv[])
         io.Fonts->Build();
     }
 
-    // Cleanup
+    // Define deinitialization procedure that is called when something in RxCpp finishes
+    // TODO: When exactly RXCPP_UNWIND_AUTO is called and why to use it?
```

Comment:
I have an article on my blog describing `UNWINDER` [blog](http://kirkshoop.github.io/2011/09/27/unwinder-should-be-in-standard-library.html)

summary - it runs the supplied lambda in a destructor of a local var. this will run when the current scope ends. avoids writing resource types for every single thing.